WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalManagerBase.java

```diff
@@ -73,4 +75,21 @@ public IntervalBase getIntervalOfType(IntervalType type) {
 		}
 		return collectedIntervals;
 	}
+
+	/**
+	 * @return An {@link ArrayList} of intervals of the specified Perspective
+	 *         type.
+	 */
+	protected List<IntervalBase> getPerspectiveIntervalsOfType(Perspective type) {
```

Comment:
getIntervalsOfType is still called from outside (mostly from IntervalStatistics class) for fetching intervals of certain type (which does not have subtypes), so making it private creates another issues. Also, calling it from other methods will again require two iterations (one over list of all intervals and another over resulting list from previous iteration, to filter specified subtype).